lib/task_listener.js

```diff
@@ -314,11 +319,102 @@ export default class TaskListener extends EventEmitter {
   }
 
   /**
+   * Coalesce the task in the given claim, claiming any additional tasks
+   * directly from the TC queue service, then calling runTask with the
+   * resulting task set.
+   */
+  async coalesceClaim(claim) {
+    let task = claim.task;
+    let taskId = claim.status.taskId;
+
+    try {
+      // if the task is not coalescible, then the taskset is just the one claim
```

Comment:
nit: doesn't need to be fixed now, but instead of reading this and caring how we determine if it's able to be coalesced, we could just have a method that returned a boolean and moved it out of here.

```js
if (!taskIsCoalescible(task)) {
  return [claim];
}
```